docs/cosmology/dev.rst

```diff
@@ -90,50 +89,49 @@ the definition of :class:`~astropy.cosmology.FLRW`.
 
 .. code-block:: python
 
-    from astropy.utils.decorators import lazyproperty
-
     class FLRW(Cosmology):
 
         H0 = Parameter(doc="Hubble constant as an `~astropy.units.Quantity` at z=0",
-                       unit="km/(s Mpc)")
-        Om0 = Parameter(doc="Omega matter; matter density/critical density at z=0")
-        Ode0 = Parameter(doc="Omega dark energy; dark energy density/critical density at z=0")
-        Tcmb0 = Parameter(doc="Temperature of the CMB as `~astropy.units.Quantity` at z=0",
-                          unit="Kelvin", fmt="0.4g")
-        Neff = Parameter(doc="Number of effective neutrino species")
-        m_nu = Parameter(doc="Mass of neutrino species",
-                         unit="eV", equivalencies=u.mass_energy(), fmt="")
-        Ob0 = Parameter(doc="Omega baryon; baryonic matter density/critical density at z=0")
+                       unit="km/(s Mpc)", fvalidate="scalar")
+        Om0 = Parameter(doc="Omega matter; matter density/critical density at z=0",
+                        fvalidate="non-negative")
+        Ode0 = Parameter(doc="Omega dark energy; dark energy density/critical density at z=0.",
+                         fvalidate="float")
+        Tcmb0 = Parameter(doc="Temperature of the CMB as `~astropy.units.Quantity` at z=0.",
+                  unit="Kelvin", fmt="0.4g", fvalidate="scalar")
+        Neff = Parameter(doc="Number of effective neutrino species.", fvalidate="non-negative")
+        m_nu = Parameter(doc="Mass of neutrino species.",
+                 unit="eV", equivalencies=u.mass_energy(), fmt="")
+        Ob0 = Parameter(doc="Omega baryon; baryonic matter density/critical density at z=0.")
 
         def __init__(self, H0, Om0, Ode0, Tcmb0=0.0*u.K, Neff=3.04, m_nu=0.0*u.eV,
                      Ob0=None, *, name=None, meta=None):
-            cls = self.__class__
-            self._H0 = H0 << cls.H0.unit
+            self.H0 = H0
             ...  # for each Parameter in turn
 
-        @m_nu.getter
-        @lazyproperty  # so only does expensive computation once
-        def m_nu(self):
-            """ Mass of neutrino species"""
+        @Ob0.validator
+        def Ob0(self, param, value):
+            """Validate baryon density to None or positive float > matter density."""
+            if value is None:
+                return value
             ...
-            return u.Quantity(numass, self.__class__.m_nu.unit)
+            return value
```

Comment:
I'd suggest to copy the whole validator, as right now it misses the point that you can access the cosmology instance easily.